Automattic/woocommerce-payments
# chore: delete old feature flags from DB

#### Changes proposed in this Pull Request









I created a migration to delete old feature flags from the DB.
I am creating it because it looks like they might have never been cleaned.
Some of the feature flags might have been internal/developer facing, but others might have been customer-facing (like the UPE ones).

Cleaning them all up.

To gather this list, I went through the history of the `includes/class-wc-payments-features.php` and all its changes.

#### Testing instructions





I added som tests for the migration.

- Insert one of the deprecated options in your DB to an older one - something like:
```sql
INSERT INTO `wp_options` (`option_name`, `option_value`, `autoload`)
VALUES ('_wcpay_feature_client_secret_encryption', '1', 'yes');
```
- Change the value of WooPayments on your DB - something like:
```sql
UPDATE `wp_options` SET
`option_value` = '7.1.1'
WHERE `option_name` = 'woocommerce_woocommerce_payments_version';
```
- Change the local version of WooPayments here to `8.1.0`: https://github.com/Automattic/woocommerce-payments/blob/develop/woocommerce-payments.php#L14-L14
- Load the WooPayments settings page
- In your DB, check that:
  - The WooPayments version changed to `8.1.0`
```sql
SELECT * FROM `wp_options` WHERE `option_name` = 'woocommerce_woocommerce_payments_version' LIMIT 50
```
  - The `_wcpay_feature_client_secret_encryption` option no longer exists
```sql
SELECT * FROM `wp_options` WHERE `option_name` = '_wcpay_feature_client_secret_encryption'
```

-------------------


**Post merge**
